spec/save_operation_spec.cr

```diff
@@ -355,6 +359,48 @@ describe "Avram::SaveOperation" do
         end
       end
     end
+
+    context "when there's default values in the table" do
+      it "saves with all of the default values" do
+        ModelWithDefaultValues::SaveOperation.create do |operation, record|
+          record.should_not eq nil
+          r = record.not_nil!
+          r.greeting.should eq "Hello there!"
+          r.admin.should eq false
+          r.age.should eq 30
+          r.money.should eq 3.5
+          r.published_at.should be_a Time
+          r.drafted_at.should be_a Time
+        end
+      end
+
+      it "allows you to override the default values" do
+        ModelWithDefaultValues::SaveOperation.create(greeting: "A fancy hat") do |operation, record|
+          record.should_not eq nil
+          r = record.not_nil!
+          r.greeting.should eq "A fancy hat"
+          r.admin.should eq false
+          r.age.should eq 30
+          r.money.should eq 3.5
+          r.published_at.should be_a Time
+          r.drafted_at.should be_a Time
+        end
+      end
+
+      it "overrides all of the defaults through params" do
+        params = Avram::Params.new({"greeting" => "Hi", "admin" => "true", "age" => "4", "money" => "100.23", "published_at" => 1.day.ago.to_s, "drafted_at" => 1.week.ago.to_s})
+        OverrideDefaults.create(params) do |operation, record|
+          record.should_not eq nil
+          r = record.not_nil!
+          r.greeting.should eq "Hi"
+          r.admin.should eq true
+          r.age.should eq 4
+          r.money.should eq 100.23
+          r.published_at.should be_a Time
+          r.drafted_at.should be_a Time
```

Comment:
```suggestion
        published_at = 1.day.ago
        drafted_at = 2.days.ago
        params = Avram::Params.new({"greeting" => "Hi", "admin" => "true", "age" => "4", "money" => "100.23", "published_at" => published_at.to_s, "drafted_at" => drafted_at.to_s})
        OverrideDefaults.create(params) do |operation, record|
          record.should_not eq nil
          r = record.not_nil!
          r.greeting.should eq "Hi"
          r.admin.should eq true
          r.age.should eq 4
          r.money.should eq 100.23
          r.published_at.should be_a published_at
          r.drafted_at.should be_a drafted_at
```

I think it'd be good to be specific about the expectation just in case it isn't setting the time correctly. You may need to do `at_beginnning_of_second` depending on the precision of the time column. Not sure

Reply:
Yeah, that one is a little tricky. In rails, there's the Timecop gem that's really nice for doing this kind of stuff. Should we look at using something like that? (provided it exists...), or would this suggested change be good enough?

Reply:
I think something like this is probably good enough (at least for now)